### PR TITLE
Enhancements

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    |  ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

- Improve readme and changelog.
- Replace phan to psalm.
- Enhancements by static analysis.
- Replace `is_callable` to `instance of \Closure` in `ArraySorter::getKeys()`.

### CODE REVIEW

File: src/ArraySorter.php [hunk 1]
```diff
@@ -75,21 +75,21 @@ public static function multisort(array &$array, $key, $direction = SORT_ASC, $so
...
-    private static function getKeys(array &$array, $key): array
+    private static function getKeys(array $array, $key): array
     {
         $keys = is_array($key) ? $key : [$key];
         if (empty($keys) || empty($array)) {
             return [];
         }
 
-        if (is_callable($key)) {
+        if ($key instanceof \Closure) {
```
Comment (samdark) [inline src/ArraySorter.php:92]: That looks suspicious. What is the reason for this change?

Comment (vjik) [reply src/ArraySorter.php:92]: ArrayHelper methods (in particular, `getColumn` that used in this method) don't support callable as array `$key = [$this, 'methodName']` or named function `$key = 'fnName'`. This is occurs error. 

Also in phpdoc for this function specified `string|\Closure|array`, not `callable`.

**Review by samdark — APPROVED** [review]
